### PR TITLE
closing a bunch of issues 

### PR DESCRIPTION
**Pull request by edublancas**
addressed a bunch of the currently opened issues. most of them are already solved so I just added tests. Also added some missing tests, and documented a few things. Commit messages contain descriptive names of what I did on each one.

**Important:** do not squash this PR, let's get all commits into the master branch since each commit contains useful messages.

### CODE REVIEW

File: doc/howto.md [hunk 1]
```diff
@@ -110,3 +110,42 @@ Pass existing engines to `%sql`
 %%sql
 SELECT MYSUM(1, 2)
 ```
+
+## Connect to a SQLite database with spaces
+
+Currently, due to a limitation in the argument parser, it's not possible to directly connect to SQLite databases whose path contains spaces; however, you can do it by creating the engine first.
```
Comment (idomic) [inline doc/howto.md:116]: We're keeping the issue open on this? It will probably be a quick fix.

File: src/tests/test_connection.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import sys
+from unittest.mock import Mock
+
+from sqlalchemy.engine import Engine
+
+from sql.connection import Connection
+
+
+def test_password_isnt_displayed(monkeypatch):
```
Comment (idomic) [inline src/tests/test_connection.py:9]: What about a test for connection with spaces?

File: src/tests/test_magic.py [hunk 3]
```diff
@@ -368,14 +370,41 @@ def test_json_in_select(ip):
...
+
+def test_column_names_visible(ip, tmp_empty):
+    res = ip.run_line_magic("sql", "SELECT * FROM empty_table")
+
+    assert "<th>column</th>" in res._repr_html_()
+    assert "<th>another</th>" in res._repr_html_()
+
+
+@pytest.mark.xfail(reason="known parse @ parser.py error")
+def test_sqlite_path_with_spaces(ip, tmp_empty):
```
Comment (idomic) [inline src/tests/test_magic.py:402]: Do we also test for windows? The issue was opened about windows, usually paths there have different behaviors than mac/linux

**Review by idomic — COMMENTED** [review]
.

**Review by idomic — CHANGES_REQUESTED** [review]
.

File: doc/connecting.md [hunk 4]
```diff
@@ -51,6 +51,39 @@ a flag with (-a|--connection_arguments)the connection string as a JSON string. S
...
+**It is highly recommended** that you do not pass plain credentials.
+
+```{code-cell} ipython3
+%load_ext sql
+```
+
+One option is to use `getpass`, type your password, build your connection string and pass it to `%sql`:
+
+```{code-cell} ipython3
+from getpass import getpass
```
[the remark this answers is not in the document]
Comment (edublancas) [reply doc/connecting.md:67]: We can provide both options, and make conf files the recommended solution. I saw in the codebase that there is support for it already, but I don't think it's well-documented. Opened https://github.com/ploomber/jupysql/issues/56

File: doc/connecting.md [hunk 5]
```diff
@@ -51,6 +51,39 @@ a flag with (-a|--connection_arguments)the connection string as a JSON string. S
...
+
+```python
+password = getpass()
+connection_string = f"postgresql://user:{password}@localhost/database"
+%sql $connection_string
+```
+
++++
+
+You may also set the `DATABASE_URL` environment variable, and `%sql` will automatically load it from there:
```
[the remark this answers is not in the document]
Comment (edublancas) [reply doc/connecting.md:78]: I'm not following, an example of what in which section?

File: doc/howto.md (hunk 1 above)
Comment (edublancas) [reply doc/howto.md:116]: It isn't a simple fix. My guess is that it's related to this: https://github.com/ploomber/jupysql/issues/33 (the parsing logic hasn't been well tested)

File: src/tests/test_connection.py (hunk 2 above)
Comment (edublancas) [reply src/tests/test_connection.py:9]: there is one already

Comment (idomic) [reply src/tests/test_connection.py:9]: Yeah I saw it later.

File: src/tests/test_magic.py (hunk 3 above)
Comment (edublancas) [reply src/tests/test_magic.py:402]: the OS is configured via github actions, not here. There's an open issue about running the tests on windows

File: doc/connecting.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (idomic) [reply doc/connecting.md:78]: Something around:

> You may also set the `DATABASE_URL` environment variable ([example here](somelink)), and `%sql` will automatically load it from there:

File: doc/connecting.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (idomic) [reply doc/connecting.md:67]: Ok.

File: src/tests/test_magic.py (hunk 3 above)
Comment (idomic) [reply src/tests/test_magic.py:402]: Ok, same, let's keep the issue open until we're certain it's done (I know Windows currently fails).

File: doc/howto.md (hunk 1 above)
Comment (idomic) [reply doc/howto.md:116]: Ok we'll keep it open until solved.

File: doc/connecting.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (edublancas) [reply doc/connecting.md]: added inline example

**Pull request merged**